src/components/structures/MessagePanel.tsx

```diff
@@ -555,18 +555,48 @@ export default class MessagePanel extends React.Component<IProps, IState> {
     private collectGhostReadMarker = (node: HTMLElement): void => {
         if (node) {
             // now the element has appeared, change the style which will trigger the CSS transition
+            let animating = false;
+            node.addEventListener('transitionrun', () => { animating = true; }, { once: true });
             requestAnimationFrame(() => {
                 node.style.width = '10%';
                 node.style.opacity = '0';
+                requestAnimationFrame(() => {
+                    if (!animating) {
+                        try {
+                            const ev = new TransitionEvent('transitionend', { bubbles: true });
+                            node.dispatchEvent(ev);
+                        } catch (err) {
+                            logger.error(err);
+                            return;
+                        }
+                    }
+                });
```

Comment:
> I'll see if I can get a demonstration, although that might be tricky because it depends upon a stream of messages from other people. 
>
> *-- https://github.com/matrix-org/matrix-react-sdk/pull/7630#issuecomment-1024365884*

We recently introduced [Cypress for end-to-end tests](https://github.com/matrix-org/matrix-react-sdk/blob/6574c5c3e2835c6cf25bed58e9e0b62a188dcf72/docs/cypress.md) to the codebase. This means you can setup a list of repeatable steps to run against a real Element interface and Synapse homeserver. This seems perfect if you want a way to demo this for real and will guarantee the behavior in the future so we don't regress.